sopel/db.py

```diff
@@ -185,6 +206,29 @@ def __init__(
                            database=db_name, query=query)
 
         self.engine = create_engine(self.url, pool_recycle=3600)
+        """SQLAlchemy Engine used to connect to Sopel's database.
+
+        .. seealso::
+
+            Read `SQLAlchemy engine`__'s documentation to know how to use it.
+
+        .. __: https://docs.sqlalchemy.org/en/14/core/connections.html
+
+        .. important::
+
+            Introduced in Sopel 7, Sopel uses SQLAlchemy 1.4+. This version of
+            SQLAlchemy deprecates various behaviors and methods, to prepare the
+            migration to its future 2.0 version, and the new 2.x style.
+
+            Sopel doesn't enforce the new 2.x style yet. This will be modified
+            in Sopel 8.1 by using the ``future=True`` flag on the engine.
```

Comment:
What about the use of `future=True` with the `sessionmaker` below? Not a compatibility concern for plugins, I take it?

Reply:
As said in the (now deprecated) `execute` method, I can't use `future=True` because of it. Once it's removed in Sopel 8.1, we should be able to set that flag.

Reply:
Set it on the engine, yes. But R244 already sets `future=True` for the `sessionmaker`. That won't cause compatibility problems for plugins, even though setting it on the engine could (because of `execute`)?

Reply:
Eeyup, that's what I tested, IIRC: one was fine, the other wasn't, because of `execute`.